charms/worker/k8s/src/charm.py

```diff
@@ -187,13 +189,16 @@ def _join_cluster(self):
         status.add(ops.MaintenanceStatus("Joining cluster"))
 
         if relation := self.model.get_relation("cluster"):
-            app_databag = relation.data.get(self.model.app, {})
+            app_databag = relation.data.get(self.model.get_app("k8s"), {})
```

Comment:
Ahh... yeah, this won't work.  The `k8s` charm can be deployed with any name the juju admin wants.  They can call it `kubernetes` or `control-plane`.  Nothing requires the name to be `k8s`.  

We may need to walk the relation.data to find the correct databag:


```suggestion
            app_databag = None
            for potential in relation.data.values():
                if self.unit.name in potential:
                    app_databag = potential
                    break
            assert app_databag, "Failed to find cluster databag"
```